src/theory/evaluator.cpp

```diff
@@ -524,6 +552,460 @@ EvalResult Evaluator::evalInternal(TNode n,
+
+          if (res.second)
+          {
+            results[currNode] = EvalResult(res.first);
+          }
+          else
+          {
+            // Can't constant fold the underspecified case
+            return EvalResult();
+          }
+
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_UBV_TOTAL:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToUBVTotal& param =
+              op.getOperator().getConst<FloatingPointToUBVTotal>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+          const BitVector& partial = results[currNode[2]].d_bv;
+
+          results[currNode] =
+              EvalResult(val.convertToBVTotal(param.bvs, rm, false, partial));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_SBV:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToSBV& param =
+              op.getOperator().getConst<FloatingPointToSBV>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+
+          FloatingPoint::PartialBitVector res(
+              val.convertToBV(param.bvs, rm, true));
+
+          if (res.second)
+          {
+            results[currNode] = EvalResult(res.first);
+          }
+          else
+          {
+            // Can't constant fold the underspecified case
+            return EvalResult();
+          }
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_SBV_TOTAL:
+        {
+          TNode op = currNode.getOperator();
+          const FloatingPointToSBVTotal& param =
+              op.getConst<FloatingPointToSBVTotal>();
+          const RoundingMode& rm = results[currNode[0]].d_rm;
+          const FloatingPoint& val = results[currNode[1]].d_fp;
+          const BitVector& partial = results[currNode[2]].d_bv;
+
+          results[currNode] =
+              EvalResult(val.convertToBVTotal(param.bvs, rm, true, partial));
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_REAL:
+        {
+          const FloatingPoint& val = results[currNode[0]].d_fp;
+
+          FloatingPoint::PartialRational res(val.convertToRational());
+
+          if (res.second)
+          {
+            results[currNode] = EvalResult(res.first);
+          }
+          else
+          {
+            // Can't constant fold the underspecified case
+            return EvalResult();
+          }
+          break;
+        }
+
+        case kind::FLOATINGPOINT_TO_REAL_TOTAL:
+        {
+          const FloatingPoint& val = results[currNode[0]].d_fp;
+          const Rational& partial = results[currNode[1]].d_rat;
+
+          results[currNode] = EvalResult(val.convertToRationalTotal(partial));
+          break;
+        }
+
+#ifdef CVC4_USE_SYMFPU
+        case kind::FLOATINGPOINT_COMPONENT_NAN:
```

Comment:
Not sure if you need to eval these?

Reply:
Probably not, I was just trying to be as complete as possible.